lib/omniauth/strategies/slack.rb

```diff
@@ -19,56 +19,37 @@ class Slack < OmniAuth::Strategies::OAuth2
         param_name: 'token'
       }
 
-      uid { raw_info['user_id'] }
+      # User ID is not guaranteed to be globally unique across all Slack users.
+      # The combination of user ID and team ID, on the other hand, is guaranteed
+      # to be globally unique.
+      uid { "#{user_identity['id']}-#{team_identity['id']}" }
 
       info do
         hash = {
-          nickname: raw_info['user'],
-          team: raw_info['team'],
-          user: raw_info['user'],
-          team_id: raw_info['team_id'],
-          user_id: raw_info['user_id']
```

Comment:
I tried to stick as much as possible to the [schema](https://github.com/intridea/omniauth/wiki/Auth-Hash-Schema#schema-10-and-later) given that one should be able to get all the extra info in the `raw_info` hash, but your comments made me realize that is not the case when only `identity` scopes are requested.

I will fix that!